src/arch/x86/reg/id.rs

```diff
@@ -1,5 +1,48 @@
+use crate::arch::RegId;
+
 // TODO: Add proper `RegId` implementation. See [issue #29](https://github.com/daniel5151/gdbstub/issues/29)
 // pub enum X86RegId {}
 
-// TODO: Add proper `RegId` implementation. See [issue #29](https://github.com/daniel5151/gdbstub/issues/29)
-// pub enum X86_64RegId {}
+/// 64-bit x86 core + SSE register identifier.
+///
+/// Source: https://github.com/bminor/binutils-gdb/blob/master/gdb/features/i386/64bit-core.xml
+/// Additionally: https://github.com/bminor/binutils-gdb/blob/master/gdb/features/i386/64bit-sse.xml
+#[derive(Debug, Clone, Copy)]
+#[non_exhaustive]
+pub enum X86_64CoreRegId {
+    /// General purpose registers: RAX, RBX, RCX, RDX, RSI, RDI, RBP, RSP, r8-r15
+    Gpr(u8),
```

Comment:
I also considered to have `Rax`, `Rbx`, etc as different enum variants but chose this design to align with the design of `X86_64CoreRegs` for now.
Which design do you prefer?

Reply:
Yep, the current design seems reasonable. 

I've typically deferred these sorts of details to the PR author, since they're more familiar with the target architecture. That said, I'm always willing to change things around (assuming you've got a good reason to prefer one over the other)

Reply:
The order of general purpose registers seems not to be well defined. 
It looks inconsistent even in the Intel SDM. For example, RSI comes before RDI in some paragraphs, but it sometimes comes after RDI in other parts:( 
That's why I had considered to have different variants for these registers. Actually, `X86CoreRegs` does so. ([crosvm](https://chromium.googlesource.com/chromiumos/platform/crosvm/+/d4ca29b5d9920de4ec387b8c89a6d30a6d4fca79/hypervisor/src/x86_64.rs#451) also does so.)

But, it's probably a different topic. The current design is also reasonable, as it simply follows the order defined in the GDB's XML file and the order is documented well. So, let me leave this design as is.

Reply:
Ahh, classic x86...

Personally, I'm now leaning towards breaking out RAX, RBX, ..., RSP into their own variants (and now that I think about it, the Segment registers as well). That said, I think it's fine if we leave the design as-is, as it matches the existing `x86[_64]CoreRegs` structure.

I'll probably open a tracking issue to break-up the GPRs + SRs into their own struct fields / variants, though that should be something I can take care of myself at some point closer to the release of `0.5`

Reply:
Sounds good!
The register names are really confusing. Actually, when I was implementing the GDB feature for crosvm, I mistakenly swapped the order of RBP and RSP, and wasted a whole day to debug it😓 
So, I'm really supportive of breaking GPRs + SRs into their own variants.

Reply:
I've gone ahead and opened the tracking issue at #35 

If you're bored, feel free to open a PR for it yourself, otherwise, I'll try to get around to it at some arbitrary point in the future :smile: